docs/cli.md

```diff
@@ -80,186 +356,217 @@ $ oc get pods -s template=production
+information that's related to a given resource.
+
+```
+$ oc describe service frontend
+```
 
-#### Examples
+### oc edit
+
+This starts an editor opened to the YAML representation of the specified object.
+When the editor exits, the object is updated.
+You can specify the editor through environment variables `OSC_EDITOR`, `GIT_EDITOR`, or `EDITOR`.
+If none of those are set, `oc edit` uses the `vi` program.
+The general form is:
 
 ```
-$ oc create -f pod.json
-$ cat pod.json | oc create -f -
-$ oc create -f http://server/pod.json
+$ oc edit <resource-type>/<name> [options]
 ```
 
-oc update
----------------
+The options are:
 
-This command can be used to update existing resources.
+| Option                      | Description                                       |
+|:----------------------------|:--------------------------------------------------|
+|`--output` (`-o`) *format*   | Edit object as *format*, one of: `json`, `yaml`.  |
+|`--output-version` *version* | Use API version *version*.                        |
 
-#### Examples
+For example, to edit the service `docker-registry` in JSON using the `v1` API format:
 
 ```
-$ oc update -f pod.json
-$ cat pod.json | oc update -f -
-$ oc update -f http://server/pod.json
+$ oc edit svc/docker-registry --output-version=v1 -o json
 ```
 
-oc delete
---------------
+### oc env
 
-This command deletes a specified resource.
+This updates the environment on a resource with a pod template.
+The general form is:
 
-#### Examples
+```
+$ oc env <resource-type>/<name> <k1>=<v1> <k2>=<v2>
+```
+
+For example:
 
 ```
-$ oc delete -f pod.json
-$ oc delete pod 1234-56-7890-234234-456456
+$ oc env dc/app DB_USER=user DB_PASS=pass
 ```
 
-oc describe
-----------------
+### oc volume
 
-This command is a wordier version of `oc get` which also integrates other
-information that's related to a given resource.
+This controls the storage associated with various resources.
+The general form is:
 
-#### Examples
+```
+$ oc volume <resource-type>/<name> --<action> <options>
+```
+
+where `<action>` is one of `add`, `remove`, `list` and `<options>` depends on `<action>`.
+For example, to arrange for the deployment configuration `registry` to access the host *_/opt_* directory, use:
 
 ```
-$ oc describe service frontend
+$ oc volume dc/registry --add --mount-path=/opt
 ```
 
-oc label
-----------------
+### oc label
 
-This command adds labels to a provided resource. 
+This command adds labels to a provided resource.
 It can also overwrite the existing labels by using the `--overwrite` flag.
 
-#### Examples
-
 ```
 $ oc label service frontend foo=bar
 ```
 
-oc stop
-----------------
-
-This command gracefully shuts down a resource by id or filename. 
-It attempts to shut down and delete a resource that supports graceful termination. 
-If the resource is resizable, it will be resized to 0 before deletion.
+### oc expose
 
-#### Examples
+Expose containers internally as services or externally via routes. There is also
```

Comment:
Start this as every other command in this doc. "This command exposes containers..."